internal/engine/interpreter/interpreter.go

```diff
@@ -212,6 +212,7 @@ type interpreterOp struct {
 	kind     wazeroir.OperationKind
 	b1, b2   byte
 	b3       bool
+	u1, u2   uint64
 	us       []uint64
```

Comment:
Do you know if the number of items in this slice is unbounded?

Reply:
it's likely to be bound to the max num of targets here:

			targets := append([]*wazeroir.BranchTargetDrop{o.Default}, o.Targets...)
			op.Rs = make([]*wazeroir.InclusiveRange, len(targets))
			op.Us = make([]uint64, len(targets))


this is emitted by the compiler at `case wasm.OpcodeBrTable:` and `numTargets` is

 		numTargets, n, err := leb128.DecodeUint32(r)

Reply:
Answering my own question, I found your comment here https://github.com/tetratelabs/wazero/pull/1296#discussion_r1150550133

It may be interesting to change us to *[16]uint64 to save space on the union type since the field is almost never used (slices take 1 pointer and 2 ints).

Reply:
Sorry I wasn't very clear. 

I think having the u1 and u2 fields is the right change. 

I was suggesting we could also improve further by changing the slice type to *[16]uint64 so we waste less space in most cases.

Reply:
yeah, the br_table instruction needs unbounded list of indexes

Reply:
Oh interesting OK 👍 

Kind of crazy but kind of cool at the same time 😁